File: Leetcode/two_pointers/75_OnePassSwapColors/Solution.java
class Solution {
    public void sortColors(int[] nums) {
       int start = 0, end = nums.length-1, firstOne=0;
       while(start<=end){
           if (nums[start]==0){
                swap(nums, start , firstOne);
                firstOne++; start++;
           }else if(nums[start]==2){
                swap(nums, start, end);
                end--;
           }else{
               start++;
           }
       }
    }
    private void swap(int[] nums, int i, int j){
        int temp = nums[i];
        nums[i] = nums[j];
        nums[j] = temp;
    }
}

File: Leetcode/two_pointers/75_OnePassSwapColors/Solution.py
class Solution(object):
    def sortColors(self, nums):
        """
        :type nums: List[int]
        :rtype: None Do not return anything, modify nums in-place instead.
        """
        lastOne, firstTwo = 0, len(nums)-1
        while not lastOne > firstTwo:
            if nums[lastOne]==1:
                lastOne+=1
            elif nums[lastOne]==0:
                nums.insert(0, nums.pop(lastOne))
                lastOne+=1
            else:
                nums.append(nums.pop(lastOne))
                firstTwo-=1
